cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java

```diff
@@ -61,17 +76,35 @@
       }
     }
     if (!topicsWithBadReplicationFactor.isEmpty()) {
-      return Collections.singleton(createTopicReplicationFactorAnomaly(topicsWithBadReplicationFactor,
+      Map<Boolean, Set<String>> topicsByFixability = new HashMap<>(2);
+      List<ConfigResource> topicResourcesToCheck = new ArrayList<>(topicsWithBadReplicationFactor.size());
+      topicsWithBadReplicationFactor.forEach(t -> topicResourcesToCheck.add(new ConfigResource(ConfigResource.Type.TOPIC, t)));
+      for (Map.Entry<ConfigResource, KafkaFuture<Config>> entry: _adminClient.describeConfigs(topicResourcesToCheck).values().entrySet()) {
+        try {
+          short topicMinISR = Short.valueOf(entry.getValue().get().get(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG).value());
+          if (topicMinISR > _targetReplicationFactor) {
+            topicsByFixability.putIfAbsent(false, new HashSet<>());
+            topicsByFixability.get(false).add(entry.getKey().name());
+          } else {
+            topicsByFixability.putIfAbsent(true, new HashSet<>());
+            topicsByFixability.get(true).add(entry.getKey().name());
+          }
+        } catch (InterruptedException | ExecutionException e) {
+          LOG.warn("Skip attempt to fix topic {}'s replication factor due to unable to retrieve its minISR config.",
+                   entry.getKey().name());
+        }
```

Comment:
Can we move this logic to a smaller function for ease of readiblity?

Reply:
I extract the logic to retrieve and cache(for following comment) minISR into a separate method